### PR TITLE
Registering Nitrogen and Carbon abundances (PR ready to be reviewed)

### PR DESCRIPTION
**Pull request by correac**
Adding the calculation of carbon and nitrogen gas abundances in the calculation of SOAP. 
This PR also includes a few suggestions in kinematic_properties.py due to error messages when running SOAP with unyt 3.0.1.

### CODE REVIEW

**Review by robjmcgibbon — CHANGES_REQUESTED** [review]
Looks good, could you please copy the change from https://github.com/SWIFTSIM/SOAP/pull/47/ to fix kinematic_properties

**Comment by correac** [comment]
@robjmcgibbon As I see it I cannot add the changes from #47 into this branch (Registering Nitrogen). This branch was checkout out from SOAP/Colibre, because the changes I propose are intended to go to SOAP/Colibre and not SOAP/Main. 
#47 was a merge request to SOAP/Main. Adding the changes from SOAP/Master into SOAP/Colibre via my current merge request is not trivial since #47 proposes changes in files that are not in SOAP/Colibre. Does this make sense?

**Comment by correac** [comment]
@robjmcgibbon Hi Rob, I am happy with the changes in this pull request. Let me know about the changes you suggested. Otherwise, great if this can be merged.

File: aperture_properties.py [hunk 1]
```diff
@@ -2118,6 +2187,150 @@ def gas_O_over_H_diffuse(self) -> unyt.unyt_array:
...
+
+        Uses a lower limit on the ratio of 1.e-4 times the solar ratio,
+        which is set in the parameter file.
+        """
+        if self.Ngas == 0:
+            return None
+        return np.log10(
+            np.clip(
+                self.gas_N_over_O_diffuse,
+                self.snapshot_datasets.get_defined_constant("N_O_sun") * 1.0e-5,
```
Comment (robjmcgibbon) [inline aperture_properties.py:2203]: Should this be 1.0e-4?

File: aperture_properties.py [hunk 2]
```diff
@@ -2118,6 +2187,150 @@ def gas_O_over_H_diffuse(self) -> unyt.unyt_array:
...
+
+        Uses a lower limit on the ratio of 1.e-4 times the solar ratio,
+        which is set in the parameter file.
+        """
+        if self.Ngas == 0:
+            return None
+        return np.log10(
+            np.clip(
+                self.gas_N_over_O_total,
+                self.snapshot_datasets.get_defined_constant("N_O_sun") * 1.0e-5,
```
Comment (robjmcgibbon) [inline aperture_properties.py:2239]: Should this be 1.0e-4?

**Review by robjmcgibbon — COMMENTED** [review]
Looks good, apart from a minor query over the limits you're using

File: aperture_properties.py (hunk 1 above)
Comment (correac) [reply aperture_properties.py]: Oh! Yes, this should be 1e-4. Fixed!

File: aperture_properties.py (hunk 2 above)
Comment (correac) [reply aperture_properties.py]: Indeed!

Comment (correac) [reply aperture_properties.py]: New version, now fixed.

**Pull request merged**